src/injected/path-snippet-controller.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { BaseStore } from '../common/base-store';
+import { PathSnippetStoreData } from '../common/types/store-data/path-snippet-store-data';
+
+export class PathSnippetController {
+    constructor(
+        private readonly pathSnippetStore: BaseStore<PathSnippetStoreData>,
+        private readonly addCorrespondingSnippet: (snippet: string) => void,
+    ) {}
+
+    public listenToStore(): void {
+        this.pathSnippetStore.addChangedListener(this.onChangedState);
+        this.onChangedState();
+    }
+
+    private onChangedState = (): void => {
+        if (this.pathSnippetStore.getState() == null) {
+            return;
+        }
+
+        const pathSnippetStoreState = this.pathSnippetStore.getState();
```

Comment:
you may move this before the `if`, avoid duplicating `this.pathSnippetStore.getState()`

Reply:
Great, yes that cleans things up :) Thanks!